solidity/random-beacon/contracts/RandomBeacon.sol

```diff
@@ -717,26 +717,35 @@ contract RandomBeacon is Ownable {
 
     /// @notice Creates a new relay entry.
     /// @param entry Group BLS signature over the previous entry.
-    function submitRelayEntry(bytes calldata entry) external {
+    /// @param groupMembers Identifiers of group members.
+    function submitRelayEntry(
+        bytes calldata entry,
+        uint32[] calldata groupMembers
+    ) external {
         uint256 currentRequestId = relay.currentRequestID;
 
         Groups.Group storage group = groups.getGroup(
             relay.currentRequestGroupID
         );
 
+        require(
+            group.membersHash == keccak256(abi.encode(groupMembers)),
+            "Invalid group members"
+        );
+
         uint256 slashingAmount = relay.submitEntry(entry, group.groupPubKey);
 
         if (slashingAmount > 0) {
-            address[] memory groupMembers = sortitionPool.getIDOperators(
-                group.members
+            address[] memory groupMembersIds = sortitionPool.getIDOperators(
+                groupMembers
```

Comment:
I just noticed that those are addresses that we are converting to, we need a rename.

```suggestion
            address[] memory groupMembersAddresses = sortitionPool.getIDOperators(
                groupMembers
```

Reply:
7e2de71